src/components/header/HeaderActions.vue

```diff
@@ -1,13 +1,33 @@
 <script setup lang="ts">
 import BaseButton from "@/components/base/BaseButton.vue";
 import Hamburger from "@/assets/icons/hamburger.svg?component";
+import { useUserStore } from "@/stores/user";
+import { useRouter } from "vue-router";
+import { computed } from "vue";
+const userStore = useUserStore();
+const router = useRouter();
+
+const loginButtonLabel = computed(() => {
+  return userStore.isLoggedIn ? "Logout" : "Login";
+});
+
+const handleLogin = () => {
+  if (userStore.isLoggedIn) {
+    userStore.logout();
+    router.push({ name: "LoginPage" });
```

Comment:
Not sure if that's what you want to do on logout, redirect to main page seems better, but best I think would be to check if we're on a page that needs authorization and if so, only then redirect to some other page. 

`route.meta `from `useRoute()` should give you access to meta tags that you normally use to determine which page should be protected. 

Right now I see you don't have protected routes so feel free to not include this part, but maybe keep that in mind :)

Reply:
Another very good suggestion. It should probably redirect to HomePage. And this will be definitely revisited on implementation of route guards :)